residency.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2017 Ondrej Smola All rights reserved.
```

Comment:
is it okay for the copyright to carry the contributor's name? We tend to use @gambol99's even for contributed files.

Really genuine question -- I'm far from being skilled on the licensing level.

Reply:
I dont care ... will update it to gambol99

Reply:
I think we should just change this to "Go-Marathon Authors" or alike now? ... we can raise a PR which changes them all

Reply:
SGTM! I'll take care of it.

Reply:
I took the liberty of changing the name myself here in order to simplify a project-wide rename later, as discussed.

Hope that's okay.